txnbuild/allow_trust_test.go

```diff
@@ -71,8 +71,46 @@ func TestAllowTrustRoundtrip(t *testing.T) {
 	allowTrust = AllowTrust{
 		SourceAccount: "MA7QYNF7SOWQ3GLR2BGMZEHXAVIRZA4KVWLTJJFC7MGXUA74P7UJVAAAAAAAAAAAAAJLK",
 		Trustor:       "GB7BDSZU2Y27LYNLALKKALB52WS2IZWYBDGY6EQBLEED3TJOCVMZRH7H",
-		Type:          CreditAsset{"USD", ""},
+		Type:          CreditAsset{"USD", "GA7QYNF7SOWQ3GLR2BGMZEHXAVIRZA4KVWLTJJFC7MGXUA74P7UJVSGZ"},
 		Authorize:     true,
 	}
+
 	testOperationsMarshallingRoundtrip(t, []Operation{&allowTrust}, true)
 }
+
+// Ensures that the issuer is back-filled from the source account correctly.
+// See https://github.com/stellar/go/pull/3636 for more context.
+func TestDecoderIncludesIssuer(t *testing.T) {
```

Comment:
Would be helpful to include the scope of the test, the type it tests, in the test name.
```suggestion
func TestAllowTrustDecoderIncludesIssuer(t *testing.T) {
```

Reply:
Isn't this implied by being in allow_trust_test? :smile:

Reply:
Functions aren't scoped by their files, so when you run the tests for the package the function name is all you got. What if two operations in the package need a TestDecoderIncludesIssuer test?